documentation/VSTestIntegration.md

```diff
@@ -0,0 +1,92 @@
+2. Authoring - When new test projects are created (dotnet mstest), references to the data collector package are added by default. This reduces adoption friction for customers.
+3. Test execution - Today Coverlet is invoked as a msbuild target and as such only works with dotnet test (and requires a csproj). With the data collector becoming available, coverage can also be collected when tests are run on built binaries (dotnet vstest)
+
+## Proposed Solution
+
+### Scenarios to support
+The following table summarizes the support that needs to be added for a seamless code coverage collection for .NET Core on both Linux and Windows platforms:
+
+| Entry point | How will code coverage be enabled? | Syntax                                                               |
+|-------------|------------------------------------|----------------------------------------------------------------------|
+|dotnet test CLI              | Through a switch to condition data collection | `dotnet test --collect:"XPlat Code Coverage"`   |
+|dotnet vstest CLI            | Through a switch to condition data collection | `dotnet vstest --collect:"XPlat Code Coverage"` |
+
+### Coverlet Options Supported with VSTest
+
+#### Default
+| Option | Summary |
+|-------------|------------------------------------|
+|Format              | Results format in which coverage output is generated. Default format is cobertura.| 
+
+#### Advanced Options (Supported via runsettings)
+These are a list of options that are supported by coverlet. These can be specified as datacollector configurations in the runsettings.
+
+| Option         | Summary                                                                                  |
+|-------------   |------------------------------------------------------------------------------------------|
+|Format          | Coverage output format. These are either cobertura, json, lcov, opencover or teamcity.   | 
+|MergeWith       | Combine the output of multiple coverage runs into a single result.                       | 
+|Exclude         | Exclude from code coverage analysing using filter expressions.                           | 
+|ExcludeByFile   | Ignore specific source files from code coverage.                                         | 
+|Include         | Explicitly set what to include in code coverage analysis using filter expressions.       | 
+|IncludeDirectory| Explicitly set which directories to include in code coverage analysis.                   |
+|SingleHit       | Specifies whether to limit code coverage hit reporting to a single hit for each location.| 
+|UseSourceLink   | Specifies whether to use SourceLink URIs in place of file system paths.                  |
+
+How to specify these options via runsettings?
+```
+<?xml version="1.0" encoding="utf-8" ?>
+<RunSettings>
+  <DataCollectionRunSettings>
+    <DataCollectors>
+      <DataCollector friendlyName="XPlat code coverage">
```

Comment:
Tech question, test runner search for a class with `DataCollectorFriendlyName` attribute named `XPlat code coverage` from output build folder, right?

Reply:
It searches for the class in various folders including the output build folder and the nuget folders.
Basically we'll call a target before VSTest that would add the nuget path of the datacollectors to the search directories(it will be picked up directly from the nuget folder)